radioactivedecay/nuclide.py

```diff
@@ -0,0 +1,651 @@
+"""
+The nuclide module defines the ``Nuclide`` class. Each ``Nuclide``
+instance represents one nuclide or element, and has a variety of ways
+for contruction based off of a nuclide or element string, atomic
+number and mass, or `zzzaaassss` canonical id. The class properties
+provide an easy way of extracting atomic number and mass information,
+as well as a clean name string. Furthermore, additional methods provide
+an access point for mass data of nuclides, and the decay data of
+radionuclides, if present in a specified dataset. The default decay
+dataset used if none is supplied to the constructor is rd.DEFAULTDATA.
+
+The code examples shown in the docstrings assume the
+``radioactivedecay`` package has been imported as:
+
+.. highlight:: python
+.. code-block:: python
+
+    >>> import radioactivedecay as rd
+
+"""
+
+from collections import deque
+from typing import Any, Dict, List, Tuple, Union
+import matplotlib
+import networkx as nx
+from radioactivedecay.decaydata import DecayData, DEFAULTDATA
+from radioactivedecay.plots import (
+    _parse_nuclide_label,
+    _parse_decay_mode_label,
+    _check_fig_ax,
+)
+from radioactivedecay.utils import parse_radionuclide
+from radioactivedecay.utils import parse_nuclide
+
+
+class Nuclide:
+    """
+    ``Nuclide`` instances serve as name and atomic number/mass parsing
+    objects for any nuclide or element.
+
+    Parameters
+    ----------
+    input : Any
+        Input value for instantiation. Can be nuclide string in name
+        format (with or without hyphen), or atomic number (Z).
+    A : int, optional
+        For atomic number as input, atomic mass can be optionally
+        added for full nuclide string construction. 
+
+    Properties
+    ----------
+    name : str
+        Nuclide string if A data is given, otherwise the element
+        symbol is returned.
+    Z : int
```

Comment:
Z -> Z_val
(Attributes should be a list of all the data held in self. variables)
& A -> A_val below